File: DotNet/CesiumLanguageWriterTests/TestVerticalOriginCesiumWriter.cs
using System.Collections.Generic;
using CesiumLanguageWriter;
using CesiumLanguageWriter.Advanced;
using NUnit.Framework;

namespace CesiumLanguageWriterTests
{
    [TestFixture]
    public class TestVerticalOriginCesiumWriter : TestCesiumPropertyWriter<VerticalOriginCesiumWriter>
    {
        [Test]
        public void VerticalOriginCanBeWrittenAsSimpleString()
        {
            const string expectedPropertyName = "foo";
            const CesiumVerticalOrigin expectedValue = CesiumVerticalOrigin.Center;

            using (OpenPacket())
            using (var writer = new VerticalOriginCesiumWriter(expectedPropertyName))
            {
                writer.Open(OutputStream);
                writer.WriteVerticalOrigin(expectedValue);
            }

            AssertExpectedJson(new Dictionary<string, object>
            {
                { expectedPropertyName, CesiumFormattingHelper.VerticalOriginToString(expectedValue) },
            });
        }

        [Test]
        public void VerticalOriginCanBeWrittenInsideInterval()
        {
            var start = new GregorianDate(2012, 6, 7, 12, 0, 0).ToJulianDate();
            var stop = start.AddSeconds(100.0);
            const string expectedPropertyName = "foo";
            const CesiumVerticalOrigin expectedValue = CesiumVerticalOrigin.Center;

            using (OpenPacket())
            using (var writer = new VerticalOriginCesiumWriter(expectedPropertyName))
            {
                writer.Open(OutputStream);
                writer.WriteInterval(start, stop);
                writer.WriteVerticalOrigin(expectedValue);
            }
            AssertExpectedJson(expectedPropertyName, new Dictionary<string, object>
            {
                { "interval", CesiumFormattingHelper.ToIso8601Interval(start, stop, Iso8601Format.Compact) },
                { VerticalOriginCesiumWriter.VerticalOriginPropertyName, CesiumFormattingHelper.VerticalOriginToString(expectedValue) },
            });
        }

        protected override CesiumPropertyWriter<VerticalOriginCesiumWriter> CreatePropertyWriter(string propertyName)
        {
            return new VerticalOriginCesiumWriter(propertyName);
        }
    }
}

File: Java/CesiumLanguageWriterTests/translatedSrc/cesiumlanguagewritertests/TestVerticalOriginCesiumWriter.java
package cesiumlanguagewritertests;


import agi.foundation.compatibility.*;
import agi.foundation.compatibility.MapHelper;
import agi.foundation.compatibility.TestContextRule;
import agi.foundation.compatibility.Using;
import cesiumlanguagewriter.*;
import cesiumlanguagewriter.advanced.*;
import java.util.Map;
import javax.annotation.Nonnull;
import org.junit.ClassRule;
import org.junit.FixMethodOrder;
import org.junit.Rule;
import org.junit.runners.MethodSorters;
import org.junit.Test;

@SuppressWarnings({
    "unused",
    "deprecation",
    "serial"
})
@FixMethodOrder(MethodSorters.NAME_ASCENDING)
public class TestVerticalOriginCesiumWriter extends TestCesiumPropertyWriter<VerticalOriginCesiumWriter> {
    @Test
    public final void verticalOriginCanBeWrittenAsSimpleString() {
        final String expectedPropertyName = "foo";
        final CesiumVerticalOrigin expectedValue = CesiumVerticalOrigin.CENTER;
        try (Using<PacketCesiumWriter> using$0 = new Using<PacketCesiumWriter>(getPacket())) {
            try (Using<VerticalOriginCesiumWriter> using$1 = new Using<VerticalOriginCesiumWriter>(new VerticalOriginCesiumWriter(expectedPropertyName))) {
                final VerticalOriginCesiumWriter writer = using$1.resource;
                writer.open(getOutputStream());
                writer.writeVerticalOrigin(expectedValue);
            }
        }
        final Map<String, Object> tempCollection$0 = MapHelper.create();
        MapHelper.add(tempCollection$0, expectedPropertyName, CesiumFormattingHelper.verticalOriginToString(expectedValue));
        assertExpectedJson(tempCollection$0);
    }

    @Test
    public final void verticalOriginCanBeWrittenInsideInterval() {
        JulianDate start = new GregorianDate(2012, 6, 7, 12, 0, 0D).toJulianDate();
        JulianDate stop = start.addSeconds(100.0);
        final String expectedPropertyName = "foo";
        final CesiumVerticalOrigin expectedValue = CesiumVerticalOrigin.CENTER;
        try (Using<PacketCesiumWriter> using$0 = new Using<PacketCesiumWriter>(getPacket())) {
            try (Using<VerticalOriginCesiumWriter> using$1 = new Using<VerticalOriginCesiumWriter>(new VerticalOriginCesiumWriter(expectedPropertyName))) {
                final VerticalOriginCesiumWriter writer = using$1.resource;
                writer.open(getOutputStream());
                writer.writeInterval(start, stop);
                writer.writeVerticalOrigin(expectedValue);
            }
        }
        final Map<String, Object> tempCollection$0 = MapHelper.create();
        MapHelper.add(tempCollection$0, "interval", CesiumFormattingHelper.toIso8601Interval(start, stop, Iso8601Format.COMPACT));
        MapHelper.add(tempCollection$0, VerticalOriginCesiumWriter.VerticalOriginPropertyName, CesiumFormattingHelper.verticalOriginToString(expectedValue));
        assertExpectedJson(expectedPropertyName, tempCollection$0);
    }

    @Override
    protected CesiumPropertyWriter<VerticalOriginCesiumWriter> createPropertyWriter(String propertyName) {
        return new VerticalOriginCesiumWriter(propertyName);
    }

    @Nonnull
    private static final TestContextRule rule$testContext = new TestContextRule();

    @Nonnull
    @Rule
    @ClassRule
    public static TestContextRule getRule$testContext() {
        return rule$testContext;
    }
}